client/src/components/molecules/RequestsTable.tsx

```diff
@@ -126,15 +147,34 @@ const RequestsTable: FunctionComponent<Props> = (props: Props) => {
         props.onChangeNumRequests!(requestsCopy.length);
         setRequests(requestsCopy);
         mutateDeleteRequest({ variables: { _id: id } });
+        props.changeRequest(-1);
     };
     const onFulfilledRequest = (request: Request) => {
         if (request._id == null) {
             return;
         }
         if (request.fulfilledAt) {
             mutateUnfulfillRequest({ variables: { _id: request._id } });
+            const targetId = request._id;
+            const tempRequests = requests.map((req) => {
+                if (req._id === targetId) {
+                    req.fulfilledAt = undefined;
+                }
+                return req;
+            });
+            const newSortedRequests = reorderRequests(tempRequests);
+            setRequests(newSortedRequests);
         } else {
             mutateFulfillRequest({ variables: { _id: request._id } });
+            const targetId = request._id;
+            const tempRequests = requests.map((req) => {
+                if (req._id === targetId) {
+                    req.fulfilledAt = new Date();
+                }
+                return req;
+            });
+            const newSortedRequests = reorderRequests(tempRequests);
+            setRequests(newSortedRequests);
```

Comment:
since this code is duplicated can we extract it and do
```
if (req._id === targetId) {
    req.fulfilledAt = request.fulfilledAt ? undefined : new Date();
}
```